src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -102,7 +102,7 @@ private void WriteManagementClientCtors(CodeWriter writer, Client client, Config
 
             using (writer.Scope())
             {
-                writer.Line($"options = new {configuration.LibraryName}ManagementClientOptions();");
+                writer.Line($"options ??= new {configuration.LibraryName}ManagementClientOptions();");
                 writer.Line($"_clientDiagnostics = new {typeof(ClientDiagnostics)}(options);");
                 writer.Line($"_pipeline = {typeof(ManagementPipelineBuilder)}.Build(tokenCredential, options);");
```

Comment:
Should we have shared variable names for things like `_pipeline` and `_clientDiagnostics` for client (class-scoped) fields? Might make sense for `options` within this method too.

Reply:
Considering the entire file is a bunch of constant strings I'm not sure if extracting all the names makes things better of just harder to read.